src/frontend/js/hooks/useUnionResource/index.ts

```diff
@@ -90,9 +91,15 @@ const useUnionResource = <
   const eofRef = useRef<Record<string, number>>(queryClient.getQueryData(eofQueryKey) ?? {});
   log('eof', eofRef.current);
 
-  const [unionQueryKey, setUnionQueryKey] = useState<string>();
+  const [unionQueryKey, setUnionQueryKey] = useState<Nullable<string>>();
+  const handleSetUnionQueryKey = (filterA: FiltersA, filterB: FiltersB) => {
+    setUnionQueryKey(
+      [JSON.stringify(filterA), JSON.stringify(filterB), new Date().getTime()].join('-'),
+    );
```

Comment:
Are you adding date into the query key to invalidate the previous query each time you are calling this method ?

If yes, you should invalidate the query key instead I think.

Reply:
It's not a react-query key.
It's juste an inner home made key in this component that trigger new fetch when it change.

Reply:
Yup but I do not still understand why you need bind datetime in this query key ?

What is the case where this method could be called but filters did not change ?

Reply:
We need the `useEffect` that containe `fetchNewPage` to be executed in order to update `useUnionResource` results.
To achive that we need to handle two corder cases that weren't done at start.
* When filters change, we update an internal key that is part of this `useEffect` dependencies.
* When a query is invalidate, we call the `reset` function that will now also update the internal key with a uniqid (timestamp).

note: previously reset rely on setting `stack` at `[]`, but this isn't enought as an invalidated query can be empty and there for stack chagne from `[]` to `[]` and the `useEffet` is not executed.